packages/dart/lib/src/storage/core_store_sem_impl.dart

```diff
@@ -30,20 +30,22 @@ class CoreStoreSembastImp implements CoreStore {
         return true;
       }());
       final Database db = await factory.openDatabase(dbPath,
-          codec: !parseIsWeb ? getXXTeaSembastCodec(password: password) : null);
+          codec: !parseIsWeb
+              ? getXXTeaSembastCodec(password: password = 'flutter_sdk')
```

Comment:
in a quick analysis, it looks like there is an error here.

even if the `password` is entered, the value will always be `flutter_sdk`

Reply:
Will check and report back

Reply:
It is a very weird the way it says `password = 'flutter_sdk'`

I would correct it to `password ?? 'flutter_sdk'`, which is probably what I meant to do.

Reply:
Good catch!

Ideally I would like to define it in the parameters, like this:
`func({String? password = 'flutter_sdk'})` (e.g. getInstance)
and I would like it to use the default value, when `func(password: null)` or when calling without arguments `func()`.

But dart does not work that way.

I will provide an example, as I found this quite confusing.

Reply:
Here is an example of how dart works (and what I do not like about it)

https://dartpad.dev/338245c6ada3f75690e2e24493387eb7?null_safety=true

Reply:
Another fix would be to mark the password as `required` and remove the default password.

But that might break some apps.